plugins/modules/ndo_port_channel_interface.py

```diff
@@ -0,0 +1,403 @@
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    state: query
+  register: query_all
+
+- name: Delete an Port Channel Interface
+  cisco.mso.ndo_port_channel_interface:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: ansible_fabric_resource_template
+    port_channel_interface: ansible_port_channel_interface
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+import copy
+import re
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import (
+    MSOModule,
+    mso_argument_spec,
+)
+from ansible_collections.cisco.mso.plugins.module_utils.template import (
+    MSOTemplate,
+    KVPair,
+)
+
+
+def lookup_valid_interfaces(interfaces):
+    interface_ids = []
+    errors_interfaces = []
+    modified_interfaces = interfaces.replace(" ", "").split(",")
+    for interface in modified_interfaces:
+        if re.fullmatch(r"((\d+/)+\d+-\d+$)", interface):
+            slots = interface.rsplit("/", 1)[0]
+            range_start, range_stop = interface.rsplit("/", 1)[1].split("-")
+            if int(range_stop) > int(range_start):
+                for x in range(int(range_start), int(range_stop) + 1):
+                    interface_ids.append("{0}/{1}".format(slots, x))
+            else:
+                errors_interfaces.append(interface)
+        elif re.fullmatch(r"((\d+/)+\d+$)", interface):
+            interface_ids.append(interface)
+        else:
+            errors_interfaces.append(interface)
+    return set(interface_ids), errors_interfaces
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True),
+        port_channel_interface=dict(type="str", aliases=["name", "port_channel"]),
+        port_channel_interface_uuid=dict(type="str", aliases=["uuid", "port_channel_uuid"]),
+        description=dict(type="str"),
+        node=dict(type="str"),
+        interfaces=dict(type="list", elements="str", aliases=["members"]),
+        interface_policy=dict(type="str", aliases=["policy", "pc_policy"]),
+        interface_policy_uuid=dict(type="str", aliases=["policy_uuid", "pc_policy_uuid"]),
+        interface_descriptions=dict(
+            type="list",
+            elements="dict",
+            options=dict(
+                interface_id=dict(type="str"),
+                description=dict(type="str"),
+            ),
+        ),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["template", "port_channel_interface"]],
+            ["state", "present", ["template", "port_channel_interface"]],
```

Comment:
isn't node and interfaces required during creation.

Reply:
only during creation not during update, that's why I added a few extra conditions when `match` is none. The question is do we want to the user to always have to input node, interfaces and policy when `state=present` which will give us a cleaner or do we want to give some flexibility to the user by separating the required attributes depending if the object is being created or updated. I always more convinced it's always better to give some flexibility to the user.